### PR TITLE
Fix isEmpty checks for embeds

### PR DESCRIPTION
**Pull request by arynxd**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [x] Other: Builder code

Closes Issue: NaN

## Description
This PR fixes some oversights for embed validation. Discord disallows embeds with no content (a space character or similar) and JDA didn't check this before sending the embed.

### CODE REVIEW

**Comment by Chew** [comment]
Shouldn't String#isBlank() be used instead which "Returns true if the string is empty or contains only white space codepoints, otherwise false."? Discord wouldn't accept white space characters

**Comment by arynxd** [comment]
Just changed to the isBlank helper, thanks for the suggestion

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 1]
```diff
@@ -156,14 +156,14 @@ public EmbedBuilder clear()
      */
     public boolean isEmpty()
     {
-        return title == null
+        return Helpers.isBlank(title)
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:159]:
```suggestion
        return (title == null || title.trim().isEmpty())
```

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 2]
```diff
@@ -156,14 +156,14 @@ public EmbedBuilder clear()
...
             && timestamp == null
             && thumbnail == null
             && author == null
             && footer == null
             && image == null
             && color == Role.DEFAULT_COLOR_RAW
-            && description.length() == 0
+            && Helpers.isBlank(description)
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:166]:
```suggestion
            && (description == null || description.toString().trim().isEmpty())
```

**Review by MinnDevelopment — COMMENTED** [review]
I think you also should update `EmbedBuilder#getLength` and `MessageEmbed#getLength` to properly check the trimmed length

**Review by arynxd — COMMENTED** [review]
Apply changes

**Comment by arynxd** [comment]
I broke my fork, so will re-open this PR

**Pull request closed without merging**